std/valgrind/index.zig

```diff
@@ -0,0 +1,367 @@
+const builtin = @import("builtin");
+const math = @import("index.zig").math;
+
+
+const noValgrind = builtin.mode == builtin.Mode.ReleaseFast or
+        builtin.mode == builtin.Mode.ReleaseSmall;
+
+
+pub fn doClientRequest(default: usize, request: usize,
+        a1: usize, a2: usize, a3: usize, a4: usize, a5: usize
+    ) usize
+{
+    if (noValgrind) {
+        return default;
+    }
+
+    switch (builtin.arch) {
+        builtin.Arch.i386 => {
+            return asm volatile (
+                \\ roll $3,  %%edi ; roll $13, %%edi
+                \\ roll $29, %%edi ; roll $19, %%edi
+                \\ xchgl %%ebx,%%ebx
+                : [_] "={edx}" (-> usize)
+                : [_] "{eax}" ([]usize{request, a1, a2, a3, a4, a5}),
+                  [_] "0" (default)
```

Comment:
This constraint code appears to be undocumented. What are you expecting it to do?

Reply:
Found a description on https://gcc.gnu.org/onlinedocs/gcc/Simple-Constraints.html#Simple-Constraints

> An operand that matches the specified operand number is allowed. If a digit is used together with letters within the same alternative, the digit should come last.

> This number is allowed to be more than a single digit. If multiple digits are encountered consecutively, they are interpreted as a single decimal integer. There is scant chance for ambiguity, since to-date it has never been desirable that ‘10’ be interpreted as matching either operand 1 or operand 0. Should this be desired, one can use multiple alternatives instead.

> This is called a matching constraint and what it really means is that the assembler has only a single operand that fills two roles which asm distinguishes. For example, an add instruction uses two input operands and an output operand, but on most CISC machines an add instruction really has only two operands, one of them an input-output operand:

> `addl #35,r12`

> Matching constraints are used in these circumstances. More precisely, the two operands that match must include one input-only operand and one output-only operand. Moreover, the digit must be a smaller number than the number of the operand that uses it in the constraint. 

I want to double check with LLVM that this is officially supported before depending on it. If not Zig should emit a compile error.